uefi/src/table/boot.rs

```diff
@@ -1625,7 +1625,7 @@ pub struct MemoryMapSize {
     pub map_size: usize,
 }
 
-/// An iterator of [`MemoryDescriptor`] that is always associated with the
+/// An array of [`MemoryDescriptor`]s that is always associated with the
```

Comment:
this is wrong.. It still returns an `MemoryMapIter` that can be indexed as well. Or am I missing sometihng?